### ISSUE TITLE
`warn-unused-ignores` doesn't work

### ISSUE THREAD

**Issue by DetachHead** [microsoft/vscode-mypy#35]
![image](https://user-images.githubusercontent.com/57028336/235559741-f39a2310-8aed-4a6e-8155-16a5ce389219.png)
# settings
```json
{
  "mypy-type-checker.importStrategy": "useBundled",
  "mypy-type-checker.args": [
    "--show-column-numbers",
    "--no-pretty",
    "--hide-error-context",
    "--warn-unused-ignores"
  ]
}
```
# logs
```
2023-05-02 11:22:45.838 [info] [Trace - 11:22:45 AM] Received notification 'window/logMessage'.
2023-05-02 11:22:45.838 [info] file:///c%3A/Users/user/project/sandpit/asdf.py :
sandpit\asdf.py:1:12:1:14: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]
sandpit\asdf.py:3: error: Unused "type: ignore" comment

2023-05-02 11:22:45.839 [info] [Trace - 11:22:45 AM] Received notification 'textDocument/publishDiagnostics'.
```

**Comment by karthiknadig** [comment]
@DetachHead In the log do you see the argument being passed to mypy?

**Comment by DetachHead** [comment]
yes
```
2023-06-16 10:09:29.417 [info] c:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\project-zbC6VfKB-py3.11\Scripts\python.exe -m mypy.dmypy --status-file C:\Users\user\AppData\Local\Temp\.vscode.dmypy_status\status-f9223072-9f9c-486d-aa66-84bf1a253973.json run -- --no-color-output --no-error-summary --show-column-numbers --show-error-code --no-pretty --show-column-numbers --no-pretty --hide-error-context --warn-unused-ignores --show-error-end c:\Users\user\project\sandpit\asdf.py
```

**Comment by karthiknadig** [comment]
for this code:
```python
foo: int = "1"

1  # type: ignore[misc]

```

This does not seem like an extension issue. I ran `mypy` from command line and I get the same:
![image](https://github.com/microsoft/vscode-mypy/assets/3840081/73e92d49-57a3-4282-8655-dc89198de162)

**Comment by DetachHead** [comment]
yeah i get the error when running it from the command line too, but the problem is that the extension does not show the unused ignore errors even though they show up when running it from the command line

![image](https://github.com/microsoft/vscode-mypy/assets/57028336/44be90bd-c2e9-4584-8654-8feee6a7ddce)

**Comment by karthiknadig** [comment]
Ah I see, It looks like the patter of output has bug in `mypy` itself. It does not give a column number. We can add a regex that can handle this case, but this looks like a bug in `mypy` itself.

**Comment by DetachHead** [comment]
i've raised it on mypy: https://github.com/python/mypy/issues/15463

however apparently that's not the only error that doesn't have column numbers, but i'm not sure what other ones there are
